pkg/csi/cinder/controllerserver.go

```diff
@@ -266,13 +285,21 @@ func (cs *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
 		klog.V(3).Infof("CreateSnapshot %s on %s", name, volumeId)
 	}
 
-	// TODO: add snapshot status
+	err = cloud.WaitSnapshotReady(snap.ID)
+	if err != nil {
+		klog.V(3).Infof("Failed to WaitSnapshotReady: %v", err)
+		return nil, err
+	}
+
 	return &csi.CreateSnapshotResponse{
 		Snapshot: &csi.Snapshot{
 			Id:             snap.ID,
 			SizeBytes:      int64(snap.Size * 1024 * 1024 * 1024),
 			SourceVolumeId: snap.VolumeID,
 			CreatedAt:      int64(snap.CreatedAt.UnixNano() / int64(time.Millisecond)),
+			Status: &csi.SnapshotStatus{
+				Type: csi.SnapshotStatus_READY,
```

Comment:
curious on why snapshot need status and volume not (so don't need wait to be available status...)

Reply:
I was also, I think we should also wait for volume creation.